Sources/App/Core/Github.swift

```diff
@@ -259,6 +259,13 @@ extension Github {
                     }
                     owner {
                       login
+                      avatarUrl
+                      ... on User {
+                        name
+                      }
+                      ... on Organization {
+                        name
+                      }
```

Comment:
Interesting, I had no idea there's an `...` operator!

I'm guessing this is sort of like an enum or union type for `name`. Is there a guarantee in the schema that there'll always be either a `User` or an `Organisation`? Or do we potentially need to make this `String?` in our `Owner` `Decodable`?

The graphql console isn't working for me at the moment so I can't check unfortunately.

Reply:
TL;DR:

- `avatarUrl` and `login` are safe to leave as `String`
- `name` should be `String?`

Oh, there're two instances of `...` in GraphQL I'm aware of:

- [GraphQL Fragments](https://graphql.org/learn/queries/#fragments) - they preserve query's locality
- [GraphQL Interfaces](https://graphql.org/learn/schema/#interfaces) and [GraphQL Unions](https://graphql.org/learn/schema/#union-types) - `... on Organization` is similar to `if let organization = owner as? Organization` to my understanding

So, long story short, there's `interface RepositoryOwner` and two instances that conform to it: `User` and `Organization`. `Repository` declares `owner: RepositoryOwner!`

- `avatarUrl: URI!` and `login: String!` are the `RepositoryOwner` interface part. Thus guaranteed to be presented
- While  `User` and `Organization` both contain the` name: String` parameter, it's not part of the `RepositoryOwner` interface

The query means "Hey if the owner typecasts to User or Organization types, get its name":

```graphql
... on User {
    name
}
... on Organization {
    name
}
```